File: check_coordinates.js
const admin = require('firebase-admin');
const serviceAccount = require('./serviceAccountKey.json');

// Firebase Admin SDK'yı başlat
admin.initializeApp({
  credential: admin.credential.cert(serviceAccount)
});

const db = admin.firestore();

async function checkCoordinates() {
  try {
    console.log('🔍 Enhanced_stops koleksiyonundaki koordinatları kontrol ediliyor...');
    
    const stopsSnapshot = await db.collection('enhanced_stops').get();
    
    let totalStops = 0;
    let invalidCoordinates = 0;
    let validCoordinates = 0;
    
    stopsSnapshot.forEach(doc => {
      const data = doc.data();
      totalStops++;
      
      const lat = data.latitude;
      const lng = data.longitude;
      
      if (lat === 0.0 || lng === 0.0 || lat == null || lng == null) {
        invalidCoordinates++;
        console.log(`❌ Geçersiz koordinat: ${doc.id} - ${data.address || 'Adres yok'} (${lat}, ${lng})`);
      } else {
        validCoordinates++;
        console.log(`✅ Geçerli koordinat: ${doc.id} - ${data.address || 'Adres yok'} (${lat}, ${lng})`);
      }
    });
    
    console.log('\n📊 ÖZET:');
    console.log(`Toplam durak: ${totalStops}`);
    console.log(`Geçerli koordinat: ${validCoordinates}`);
    console.log(`Geçersiz koordinat: ${invalidCoordinates}`);
    
    if (invalidCoordinates > 0) {
      console.log('\n🔧 Geçersiz koordinatları düzeltmek için geocoding servisi çalıştırılmalı.');
    }
    
  } catch (error) {
    console.error('❌ Hata:', error);
  } finally {
    process.exit(0);
  }
}

checkCoordinates();

File: check_coordinates.dart
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:firebase_core/firebase_core.dart';

void main() async {
  await Firebase.initializeApp();

  print('🔍 Enhanced_stops koleksiyonundaki koordinatları kontrol ediliyor...');

  try {
    final stopsSnapshot =
        await FirebaseFirestore.instance.collection('enhanced_stops').get();

    print('📊 Toplam durak sayısı: ${stopsSnapshot.docs.length}');
    print('\n--- KOORDİNAT KONTROLÜ ---');

    int zeroCoordinateCount = 0;
    int validCoordinateCount = 0;

    for (var doc in stopsSnapshot.docs) {
      final data = doc.data();
      final name = data['name'] as String?;
      final latitude = data['latitude'] as double?;
      final longitude = data['longitude'] as double?;
      final address = data['address'] as String?;
      final isActive = data['isActive'] as bool?;

      print('Durak: $name');
      print('  - ID: ${doc.id}');
      print('  - Adres: $address');
      print('  - Latitude: $latitude');
      print('  - Longitude: $longitude');
      print('  - Aktif: $isActive');

      if (latitude == null ||
          longitude == null ||
          latitude == 0.0 ||
          longitude == 0.0) {
        print('  ❌ SORUNLU KOORDİNAT!');
        zeroCoordinateCount++;
      } else {
        print('  ✅ Geçerli koordinat');
        validCoordinateCount++;
      }
      print('---');
    }

    print('\n📈 SONUÇ:');
    print('✅ Geçerli koordinatlı durak sayısı: $validCoordinateCount');
    print('❌ Sorunlu koordinatlı durak sayısı: $zeroCoordinateCount');

    if (zeroCoordinateCount > 0) {
      print('\n🔧 Koordinat sorunu tespit edildi!');
      print('Bu durakların koordinatları düzeltilmeli.');
    } else {
      print('\n✅ Tüm durakların koordinatları geçerli.');
    }

    print('\n✅ Kontrol tamamlandı!');
  } catch (e) {
    print('❌ Hata oluştu: $e');
  }
}
